libgadget/exchange.c

```diff
@@ -113,8 +113,8 @@ int domain_exchange(ExchangeLayoutFunc layoutfunc, const void * layout_userdata,
         }
 
         /* determine for each rank how many particles have to be shifted to other ranks */
-        plan.last = domain_find_iter_space(&plan);
-        domain_build_plan(layoutfunc, layout_userdata, &plan);
+        plan.last = domain_find_iter_space(&plan, PartManager, SlotsManager);
```

Comment:
shall we move part manager and slots manager into plan?

Reply:
We could but I somehow don't like the look of it, if that's ok.

Reply:
Sure; it can be beneficial to keep interfaces small. On the other hand, if we always use PartMan and SlotMan together then probably will want to merge them into the same interface -- something that models a table with extensible rows (variable number of columns per row) span across different MPI ranks.